x/gamm/pool-models/balancer/pool_suite_test.go

```diff
@@ -629,169 +788,12 @@ func (suite *KeeperTestSuite) TestBalancerSpotPriceBounds() {
-			// P_issued = P_supply * ((1 + (A_t * swapFeeRatio  / B_t))^W_t - 1)
-			// 609,374,990,000 = (1e20 + 1,250,000,000,000) *  (( 1 + (37,500 * (1 - (1 - 1/6) * 0.03) / 10,000,00,025,000))^1/6 - 1)
-			//
-			// where:
-			// 	P_supply = initial pool supply = 1e20 + 1_250_000_000_000 (from first join pool)
-			//	A_t = amount of deposited asset = 37,500
-			//	B_t = existing balance of deposited asset in the pool prior to deposit = 1,000,000,025,000
-			//	W_t = normalized weight of deposited asset in pool = 0.5 (equally weighted two-asset pool)
-			// 	swapFeeRatio = (1 - (1 - W_t) * swapFee)
-			// Plugging all of this in, we get:
-			// 	Full solution with fees: https://www.wolframalpha.com/input?i=%28100+*10%5E18+%2B+1250000000000%29*%28%281%2B++++%2837500*%281+-+%281-1%2F6%29+*+0.03%29%2F1000000012500%29%29%5E%281%2F6%29+-+1%29
-			// 	Simplified:  P_issued = 1,250,000,000,000 + 609,374,990,000
-			name:    "Multi-tokens In: unequal amounts, with unequal weights with 0.03 swap fee",
-			swapFee: sdk.MustNewDecFromStr("0.03"),
-			poolAssets: []balancer.PoolAsset{
-				{
-					Token:  sdk.NewInt64Coin("uosmo", 2_000_000_000_000),
-					Weight: sdk.NewInt(500),
-				},
-				defaultAtomPoolAsset,
-			},
-			tokensIn: sdk.NewCoins(
-				sdk.NewInt64Coin("uosmo", 25_000),
-				sdk.NewInt64Coin("uatom", 50_000),
-			),
-			expectShares: sdk.NewInt(1250000000000 + 609374990000),
-		},
-		{
-			// This test doubles the liquidity in a fresh pool, so it should generate the base number of LP shares for pool creation as new shares
-			// This is set to 1e20 (or 100 * 10^18) for Osmosis, so we should expect:
-			// P_issued = 1e20
-			name:    "minimum input with two assets and minimum liquidity",
-			swapFee: sdk.MustNewDecFromStr("0"),
-			poolAssets: []balancer.PoolAsset{
-				{
-					Token:  sdk.NewInt64Coin("uosmo", 1),
-					Weight: sdk.NewInt(100),
-				},
-				{
-					Token:  sdk.NewInt64Coin("uatom", 1),
-					Weight: sdk.NewInt(100),
-				},
-			},
-			tokensIn: sdk.NewCoins(
-				sdk.NewInt64Coin("uosmo", 1),
-				sdk.NewInt64Coin("uatom", 1),
-			),
-			expectShares: sdk.NewInt(1e18).Mul(sdk.NewInt(100)),
-		},
-		{
-			// Pool liquidity is changed by 1e-12
-			// P_issued = 1e20 * 1e-12 = 1e8
-			name:    "minimum input two assets equal liquidity",
-			swapFee: sdk.MustNewDecFromStr("0"),
-			poolAssets: []balancer.PoolAsset{
-				{
-					Token:  sdk.NewInt64Coin("uosmo", 1_000_000_000_000),
-					Weight: sdk.NewInt(100),
-				},
-				{
-					Token:  sdk.NewInt64Coin("uatom", 1_000_000_000_000),
-					Weight: sdk.NewInt(100),
-				},
-			},
-			tokensIn: sdk.NewCoins(
-				sdk.NewInt64Coin("uosmo", 1),
-				sdk.NewInt64Coin("uatom", 1),
-			),
-			expectShares: sdk.NewInt(100_000_000),
-		},
-	}
-	testCases = append(testCases, calcSingleAssetJoinTestCases...)
+
+	testCases := append(joinPoolTestCases, calcSingleAssetJoinTestCases...)
```

Comment:
The former tests multi-assets inputs and edge cases, while the latter only tests single asset inputs. 

Splitting them up like this allows us to minimize repeated test cases as not all cases are applicable to all functions we're testing

Reply:
wdyt about splitting joinPoolTestCases into multi asset inputs and edge cases?
I personally think separating these out into two different testing varialbes would be a cleaner approach + less confusing when we try to use these test cases in other cases in the future, since each var would have a sole and clear purpose. lmk what you think!

Reply:
Since all the edge case tests in `joinPoolTestCases` are multi-asset I just renamed it to `multiAssetInputTestCases` – does this address your point?

Reply:
Yeah this sgtm!